src/physics/water_simple.f90

```diff
@@ -96,7 +102,37 @@ subroutine water_simple(sst, psfc, wind, ustar, qv, temperature,  &
             do i=2,nx-1
-                if (landmask(i,j)==kLC_WATER) then
+                ! if (landmask(i,j)==kLC_WATER) then
+                ! if (options%lsm_options%lake_category/=-1) then
+                if(                                                                         &
+                    ( (options%physics%watersurface==kWATER_SIMPLE) .AND.                   &   ! If lakemodel is not selected, use this
+                      (landmask(i,j)==kLC_WATER)                                            & !(n.b. in case noah (mp or lsm) is not used, landmask may not be set correctly!)
+                    ! ((vegtype(i,j).eq.options%lsm_options%water_category) .OR.              &   !   for every gridcell that is water
+                    ! (vegtype(i,j).eq.options%lsm_options%lake_category))                    &   !   or lake
+                    )                                                                       &
+                    .OR.                                                                    &
+                    ( (options%physics%watersurface==kWATER_LAKE) .AND.                     &    ! if lake model is selected, and
+                      (vegtype(i,j).eq.options%lsm_options%water_category) .AND.            &    !   gridcell is water,
+                      (vegtype(i,j).ne.options%lsm_options%lake_category)                   &    !   but not lake (i.e ocean)
+                    )                                                                       &
+                    ! .OR.  &
+                    ! ((options%physics%watersurface==kWATER_LAKE) .AND.                      &    ! if lake model is selected, and
+                    ! ! (landmask(i,j)==kLC_WATER) .and.                                       &
+                    !     ((vegtype(i,j).eq.options%lsm_options%water_category) .AND.         &    !   gridcell is water,
+                    !      (vegtype(i,j).ne.options%lsm_options%lake_category) .AND.          &    !   but not lake (i.e ocean)
+                    !      (lakeflag/=0)                                                      & ! if lakeflag=0, then lake_cat=-1 and the simple_water and lake-model are both run for lake cells.
+                    !     )                                                                   &
+                    !     .OR.                                                                & !
+                    !     (lakeflag==0 .AND.                                                  &
+                    !      vegtype(i,j).eq.options%lsm_options%water_category .AND.          &
+                    !      hgt(i,j)<lake_min_elev                                             &
+                    !     )                                                                   &
+                        ! If no lake cat is provided (lakeflag=0), the lake model will determine lakes based on the criterion (ivgtyp(i,j)==iswater .and. ht(i,j)>=lake_min_elev))
```

Comment:
can this all be removed now?

Reply:
Yes, unless we want it to be possible to run without a landmask but with a vegtype (in the hi-res file), which I think is a rather unlikely scenario.